prover/scripts/pcs_benchmark.rs

```diff
@@ -0,0 +1,229 @@
+use clap::Parser;
+
+#[derive(Parser)]
+#[command(name = "MyApp")]
+#[command(version = "1.0")]
+#[command(about = "Does awesome things", long_about = None)]
+struct Cli {
+    #[arg(long)]
+    columns: usize,
+    #[arg(long)]
+    log2_rows: usize,
+    #[arg(long)]
+    num_runs: usize,
+}
+
+use core::num;
+use itertools::Itertools;
+use p3_challenger::CanObserve;
+use p3_challenger::FieldChallenger;
+use p3_commit::Pcs;
+use p3_commit::TwoAdicMultiplicativeCoset;
+use p3_field::AbstractField;
+use p3_field::PrimeField32;
+use p3_matrix::dense::RowMajorMatrix;
+use rand::rngs::OsRng;
+use sp1_core::utils::inner_fri_config;
+use sp1_core::utils::inner_perm;
+use sp1_core::utils::InnerChallenge;
+use sp1_core::utils::InnerChallenger;
+use sp1_core::utils::InnerCompress;
+use sp1_core::utils::InnerDft;
+use sp1_core::utils::InnerHash;
+use sp1_core::utils::InnerPcs;
+use sp1_core::utils::InnerPcsProof;
+use sp1_core::utils::InnerVal;
+use sp1_core::utils::InnerValMmcs;
+use sp1_recursion_compiler::config::InnerConfig;
+use sp1_recursion_compiler::ir::Array;
+use sp1_recursion_compiler::ir::Builder;
+use sp1_recursion_compiler::ir::Usize;
+use sp1_recursion_compiler::ir::Var;
+use sp1_recursion_core::runtime::Runtime;
+use sp1_recursion_core::runtime::DIGEST_SIZE;
+use sp1_recursion_program::challenger::CanObserveVariable;
+use sp1_recursion_program::challenger::DuplexChallengerVariable;
+use sp1_recursion_program::challenger::FeltChallenger;
+use sp1_recursion_program::fri::types::TwoAdicPcsRoundVariable;
+use sp1_recursion_program::fri::TwoAdicMultiplicativeCosetVariable;
+use sp1_recursion_program::hints::Hintable;
+use std::cmp::Reverse;
+
+use sp1_recursion_program::commit::PcsVariable;
+use sp1_recursion_program::fri::TwoAdicFriPcsVariable;
+use sp1_recursion_program::reduce::const_fri_config;
+
+fn main() {
+    // cargo run --release -p sp1-prover --bin pcs-benchmark -- --columns 100 --log2-rows 19 --num-runs 10
+    // 1) read column / rows from command line
+    // 2) generate mock proof
+    // 3) run pcs commit and track how much time it takes
+    let cli = Cli::parse();
+
+    let num_cols = cli.columns;
+    let log2_rows = cli.log2_rows;
+
+    println!("Warming up proof...");
+    for _ in 0..3 {
+        full_proof(num_cols, log2_rows);
+    }
+    println!("Starting run of proof...");
+    let now = std::time::Instant::now();
+    for _ in 0..cli.num_runs {
+        full_proof(num_cols, log2_rows);
+    }
```

Comment:
here, you don't need to test all the other stuff, u kinda just need to test pcs_commit